### DISCUSSION TITLE
PNG Encoding is slower than .Net Framework System.Windows.Media with the same compression level

### DISCUSSION THREAD

**Discussion by ratok-mk** [SixLabors/ImageSharp#2010] [Ideas]
**A performance comparison of 100 times encoding and decoding of 2560 x 1920 PNG 16bit greyscale image on Intel Core i7 8850H took:**
System.Windows.Media: ~30 seconds
ImageSharp BestSpeed: ~34 seconds
ImageSharp DefaultCompression: ~60 seconds

**Debugging showed that decoding was close together, but encoding took far more time for ImageSharp.**

**Comment by JimBobSquarePants** [comment]
No. If it’s a problem for you. Look into it yourself.

**Reply by ratok-mk** [reply]
I will give it a try. All hints are welcome, where to look first.

**Reply by JimBobSquarePants** [reply]
We're using a custom `DeflateStream` implementation for Png encoding because the runtime `DeflateStream` implementation does not allow setting the compression range or properly and also failed to compress certain pixel layouts well. 

Before doing any work I would create a proper benchmark (using BenchMarkDotNet) to ensure that you are comparing like-for-like compression. There's a lot more to encoding than simply comparing compression speed. If System.Windows.Media is anything like System.Drawing there's generally about 15% difference in file output size between ourselves an them.
